core/discovery/brokerdiscovery/storage.go

```diff
@@ -121,12 +121,16 @@ func (s *ProposalStorage) GetProposal(id market.ProposalID) (*market.ServiceProp
 	return &s.proposals[index], nil
 }
 
-// AddProposal appends given proposal to storage
+// AddProposal appends given proposal to storage. If proposal is not supported it will not be added.
 func (s *ProposalStorage) AddProposal(proposals ...market.ServiceProposal) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
 	for _, p := range proposals {
+		if !p.IsSupported() {
```

Comment:
I would not suggest to add logic it to storage level, it's fetcher responsibility to validate proposal before adding.
+Simple fetcher condition before adding and those complex mocked tests won't be needed